great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -722,6 +729,29 @@ def _create_client(self):
 
         return boto3.client("s3", **self.boto3_options)
 
+    def _assume_role(self):
+        import boto3
+
+        # create an STS client object that represents a live connection to the
+        # STS service
+        sts_client = boto3.client("sts", **self.boto3_options)
+
+        logger.debug(f"Assuming role {self.role_arn}...")
+
+        # Call the assume_role method of the STSConnection object and pass the role
+        # ARN and a role session name.
+        assumed_role_object = sts_client.assume_role(
+            RoleArn=self.role_arn, RoleSessionName="AssumedRoleSession"
+        )
+
+        # From the response that contains the assumed role, get the temporary
+        # credentials that can be used to make subsequent API calls
+        self._boto3_options["aws_access_key_id"] = assumed_role_object["Credentials"]["AccessKeyId"]
+        self._boto3_options["aws_secret_access_key"] = assumed_role_object["Credentials"]["SecretAccessKey"]
+        self._boto3_options["aws_session_token"] = assumed_role_object["Credentials"]["SessionToken"]
+
+        logger.debug(f"Role {self.role_arn} assumed, access keys configured.")
```

Comment:
Did you mean to put `self.role_assumed = True` somewhere in this `_assume_role` method? If not, feel free to delete the variable from the `__init__` method

Reply:
oh yes in a first version I used that.. was left there unintentionally. will delete, thanks!

Reply:
Sorry for the delay @ext-gvillafane. Can you pull latest develop into this branch?